src/components/PendingTxsSnackbar/store.test.js

```diff
@@ -0,0 +1,317 @@
+import { TransactionType, TransactionStatus } from 'constants';
+import { observable } from 'mobx';
+import PendingTxsSnackbarStore, { INIT_VALUES } from './store';
+import { mockResetTransactionList, mockAddTransaction } from '../../network/graphql/queries/';
+
+jest.mock('../../network/graphql/queries'); // block and manually mock our backend
+
+describe('PendingTxsSnackbarStore', () => {
+  let store;
+  const app = {
+    global: observable({ syncBlockNum: 0, syncPercent: 10 }),
+  }; // mock the appstore
+  const sleep = ms => new Promise(res => setTimeout(res, ms));
+
+  describe('Init', () => {
+    beforeEach(() => {
+      store = new PendingTxsSnackbarStore(app); // create a new instance before each test case
+      mockResetTransactionList();
+    });
+
+    it('Init with null tx list', async () => {
+      await store.init();
+      expect(store.count).toBe(INIT_VALUES.count);
+      expect(store.isVisible).toBe(INIT_VALUES.isVisible);
+      expect(store.pendingCreateEvents).toEqual(INIT_VALUES.pendingCreateEvents);
+      expect(store.pendingBets).toEqual(INIT_VALUES.pendingBets);
+      expect(store.pendingSetResults).toEqual(INIT_VALUES.pendingSetResults);
+      expect(store.pendingVotes).toEqual(INIT_VALUES.pendingVotes);
+      expect(store.pendingFinalizeResults).toEqual(INIT_VALUES.pendingFinalizeResults);
+      expect(store.pendingWithdraws).toEqual(INIT_VALUES.pendingWithdraws);
+      expect(store.pendingTransfers).toEqual(INIT_VALUES.pendingTransfers);
+      expect(store.pendingResetApproves).toEqual(INIT_VALUES.pendingResetApproves);
+      expect.assertions(10);
+    });
+  });
+
+  describe('queryPendingTransactions()', () => {
+    beforeEach(() => {
+      store = new PendingTxsSnackbarStore(app); // create a new instance before each test case
+      mockResetTransactionList();
+    });
+
+    it('Fetches all the pending txs and increments the counts', async () => {
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(0);
+      expect(store.isVisible).toBe(false);
+
+      mockAddTransaction({ txid: 1, status: TransactionStatus.PENDING, type: TransactionType.APPROVE_CREATE_EVENT });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(1);
+      expect(store.pendingBets.length).toBe(0);
```

Comment:
although its not a bad practice to check for any side effects, its not necessary to check this. since we know that we are adding an `ApproveCreateEvent tx`, we only really are concerned that`pendingCreateEvents` gets incremented. so checking all the other types is not necessary unless they have already been incremented (from a previous mock add tx). 

think cause and effect.

Reply:
Yes I think you are right. It's too duplicated and many of them can be removed.

But I'm still a little worried about whether we accidentally increased those incorrect params.
Should we add more testing cases like "Fetches a create event txs and check whether the correct param get increased"?